### PR TITLE
Controllers specs

### PR DESCRIPTION
**Pull request by umair-rao**
In this PR I have implemented the following, you are requested to please review:

✅Add RSpec as a gem to your project.
✅Create a Request spec file for all your controllers.
✅Make sure that for each action you check:
✅If the response status was correct.
✅If a correct template was rendered.
✅If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  describe 'Get /posts/:id' do
+    it 'renders show templates' do
+      get '/posts/id'
+      expect(response).to have_http_status(:success)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('Posts Details')
+    end
+  end
```
Comment (Munsa1) [inline spec/requests/posts_spec.rb]: - [ ] Great job adding your tests, however kindly ensure that they are all passing.

screenshot
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @umair-rao, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile.lock [hunk 2]
```diff
@@ -300,6 +304,7 @@ DEPENDENCIES
   pg (~> 1.1)
   puma (>= 5.0)
   rails (~> 7.1.2)
+  rails-controller-testing
   rspec-rails (~> 5.0)
```
Comment (Whoistolu) [inline Gemfile.lock:308]: - [ ] [OPTIONAL] The `rspec-rails` gem is supposed to be placed in the 
``` ruby
group :development, :test do
 gem 'rspec-rails'
end
```

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/requests/posts_spec.rb]: - Well resolved
Screenshot 2024-01-20 at 09 11 05

**Review by Whoistolu — APPROVED** [review]
### Hi @umair-rao 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

NB: I am adding this suggestion here because the `gem` in question is not in the file changes! 

- [ ] [OPTIONAL] The `rspec-rails` gem is supposed to be placed in the 
``` ruby
group :development, :test do
 gem 'rspec-rails'
end
```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**